### PR TITLE
Refactor 6LoWPAN function

### PR DESCRIPTION
**Pull request by thvdveld**
Now, handling UDP packets is done using `process_udp`. Also removed `cfg_if!` in `interface` since that messed with formatting of the code.

It also contains a fix for the fragmentation.

### CODE REVIEW

**Comment by benbrittain** [comment]
Can confirm this fixes the 6LoWPAN fragmentation errors.

File: src/iface/interface.rs [hunk 1]
```diff
@@ -778,6 +778,33 @@ impl<'a> IpPacket<'a> {
     }
 }
 
+/// Struct only used internally. We want to be able to use `process_udp` for every medium we
```
Comment (Dirbaio) [inline src/iface/interface.rs]: This sounds very very complicated. Why not pass the parsed UdpRepr (src+dst ports) and the UDP payload to `process_udp` instead? This way each caller can do the parsing accordingly. (if there's multiple callers to `process_udp` uncompressed, perhaps having a `process_ip_udp` fn could help avoid duplicating parsing code?).

Comment (thvdveld) [reply src/iface/interface.rs]: I changed it. But I'm still not happy about it. The ICMP reply for when the destination is unreachable makes everything more difficult.

Comment (thvdveld) [reply src/iface/interface.rs]: I guess it always has been incorrect for 6LoWPAN.

**Comment by thvdveld** [comment]
Going to merge this, but making an issue for the ICMP unreachable payload that is incorrect for 6LoWPAN (which has always been the case since for now).

bors r+

**Comment by thvdveld** [comment]
Apparently ICMPv6 unreachable error messages are created correctly, and I just thought they were wrong.

**Pull request merged**